frontend/app/[team]/settings/page.tsx

```diff
@@ -16,21 +17,29 @@ import { FaMoon, FaSun } from 'react-icons/fa'
 import Spinner from '@/components/common/Spinner'
 
 export default function Settings({ params }: { params: { team: string } }) {
+  const searchParams = useSearchParams()
+  const initialTabName = searchParams?.get('tab') ?? 'organisation'
+
   const { activeOrganisation } = useContext(organisationContext)
 
   const { data: session } = useSession()
 
   const activeUserIsAdmin = activeOrganisation ? userIsAdmin(activeOrganisation.role!) : false
 
-  const tabList = () => [
+  const tabList = [
     ...(activeUserIsAdmin ? [{ name: 'Organisation' }] : []),
-    ...[{ name: 'Account' }, { name: 'App' }],
+    { name: 'Account' },
+    { name: 'App' },
   ]
 
+  const initialTabIndex = tabList.findIndex(
+    tab => tab.name.toLowerCase() === initialTabName.toLowerCase()
+  )
+
```

Comment:
This works for the most part, except when already on the settings page but a different tab (since the settings page component is already mounted):

https://github.com/user-attachments/assets/e76cd7c2-11e7-4f1c-a68b-295d0b8c21a5

You may need to use a state variable for `tabName`, and use a `useEffect` hook on `searchParams` to update it:

```ts
const [tabName, setTabName] = useState('')
  
useEffect(() => {
    const initialTabName = searchParams?.get('tab') ?? 'organisation'
    setTabName(initialTabName)
}, [searchParams])
```

Then use the `tabName` instead of `initialTabName` to determine the `initialTabIndex`:

```ts
const initialTabIndex = tabList.findIndex(
    tab => tab.name.toLowerCase() === tabName.toLowerCase()
  )
```